### ISSUE TITLE
Missing support for old-style coroutines

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#2868]
Pyright doesn't support old-style coroutines (https://docs.python.org/3/library/asyncio-task.html#asyncio.coroutine):

```
@asyncio.coroutine
def old_style_coroutine():
    yield from asyncio.sleep(1)  # "Coroutine[Any, Any, _T@sleep]" is not iterable

async def main():
    await old_style_coroutine()  # "Generator[Unknown, None, None]" is not awaitable
```

(Pylance v2022.1.1)

This mechanism is deprecated and will be removed in 3.11, so there's a good case to be made against adding support for it, but thought I'd note it for completeness.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.210, which I just published. It will also be included in the next release of pylance.
